dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -218,4 +236,286 @@ public Response deleteEnvironmentPushHistory(@Context HttpServletRequest request
+                                                final Locale locale) {
+
+        final String message = Try.of(()->LanguageUtil.get(locale, "bundle.deleted.success.msg", bundleDeletesSize))
+                .onFailure(e -> Logger.error(this, e.getMessage()))
+                .getOrElse(bundleDeletesSize + " Bundles Deleted Successfully");
+
+        this.systemMessageEventUtil.pushMessage(new SystemMessageBuilder()
+                .setMessage(message)
+                .setLife(DateUtil.SEVEN_SECOND_MILLIS)
+                .setSeverity(MessageSeverity.INFO).create(), Collections.singletonList(initData.getUser().getUserId()));
+    }
+
+    // one transaction for each bundle
+    private void deleteBundleByIdentifier(final DeleteBundlesByIdentifierForm deleteBundlesByIdentifierForm, final InitDataObject initData) throws DotDataException {
+
+        for (final String bundleId : deleteBundlesByIdentifierForm.getIdentifiers()) {
+
+            this.bundleAPI.deleteBundleAndDependencies(bundleId, initData.getUser());
+        }
+    }
+
+    /**
+     * Deletes bundles older than a date. (unsent are not going to be deleted)
+     * Note: the response will be notified by socket message
+     * @param request   {@link HttpServletRequest}
+     * @param response  {@link HttpServletResponse}
+     * @param olderThan {@link ISODateParam} an ISO date, should be before now to be valid
+     */
+    @DELETE
+    @Path("/olderthan/{olderThan}")
+    @Produces("application/json")
+    public Response deleteBundlesOlderThan(@Context   final HttpServletRequest request,
+                                       @Context   final HttpServletResponse response,
+                                       @PathParam("olderThan") final ISODateParam olderThan) {
+
+        if(olderThan.after(new Date())) {
+
+            throw new IllegalArgumentException("To avoid deleting bundles that publish in the future, the date can not be after the current date");
+        }
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        final Locale locale = LocaleUtil.getLocale(request);
+
+        Logger.info(this, "Deleting the bundles older than: " + olderThan
+                + " by the user: " + initData.getUser().getUserId());
+
+        final DotSubmitter dotSubmitter = DotConcurrentFactory
+                .getInstance().getSubmitter(BUNDLE_THREAD_POOL_SUBMITTER_NAME);
+        dotSubmitter.execute(() -> {
+
+            try {
+
+                final int bundleDeletedSize =
+                        this.bundleAPI.deleteBundleAndDependenciesOlderThan(olderThan, initData.getUser()).size();
+
+                this.sendSuccessDeleteBundleMessage(bundleDeletedSize, initData, locale);
+            } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)